services/bonita-classloader/bonita-classloader-impl/src/main/java/org/bonitasoft/engine/classloader/ClassLoaderServiceImpl.java

```diff
@@ -262,4 +265,30 @@ public void resume() {
         }
         // Nothing to do
     }
+
+    @Override
+    public boolean addListener(ClassLoaderListener classLoaderListener) {
+        VirtualClassLoader localClassLoader = getVirtualClassLoaderOnContext();
+        return localClassLoader.addListener(classLoaderListener);
+    }
+
+    @Override
+    public boolean addListener(String type, long id, ClassLoaderListener classLoaderListener) {
+        final VirtualClassLoader localClassLoader = ((VirtualClassLoader) getLocalClassLoader(type, id));

```

Comment:
We can change the return type of getLocalClassLoader only in the implementation to avoid the cast.